resources/test/wptserver.py

```diff
@@ -1,48 +1,51 @@
 import json
 import os
-import ssl
 import subprocess
+import time
 import urllib2
 
 _CONFIG_FILE = os.path.join(os.path.dirname(os.path.abspath(__file__)),
                             'config.test.json')
 
-with open(_CONFIG_FILE, 'r') as config_handle:
-    config = json.loads(config_handle.read())
-    host = config["host"]
-    port = config["ports"]["https"][0]
 
 class WPTServer(object):
-    base_url = 'https://%s:%s' % (host, port)
 
     def __init__(self, wpt_root):
         self.wpt_root = wpt_root
+        with open(_CONFIG_FILE, 'r') as config_handle:
+            config = json.load(config_handle)
+        self.host = config["host"]
+        self.http_port = config["ports"]["http"][0]
+        self.https_port = config["ports"]["https"][0]
+        self.base_url = 'http://%s:%s' % (self.host, self.http_port)
+        self.https_base_url = 'https://%s:%s' % (self.host, self.https_port)
 
     def start(self):
         self.devnull = open(os.devnull, 'w')
         self.proc = subprocess.Popen(
             [os.path.join(self.wpt_root, 'wpt'), 'serve', '--config=' + _CONFIG_FILE],
-            stdout=self.devnull,
             stderr=self.devnull,
             cwd=self.wpt_root)
-        context = ssl.SSLContext(ssl.PROTOCOL_TLSv1)
-        context.verify_mode = ssl.CERT_NONE
-        context.check_hostname = False
 
-        while True:
+        # Exponential backoff.
+        wait = 1
+        for _ in range(5):
+            time.sleep(wait)
```

Comment:
If we move this to the end of the loop body, we might avoid unnecessary startup delays. What do you think?

Reply:
Also, `time.sleep(_ ** 2)` might make your intention more clear (along with renaming `_` and removing `wait`)

Reply:
I actually put it here intentionally as I noticed wptserve did take a short while to start and the first attempt often failed.

You meant `2 ** iterations` right? Yeah that'd be a bit cleaner. Let me change it.